src/components/pagination/pagination-button.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  StandaloneCircleFillIcon,
+  StandaloneEllipsisRegularIcon,
+} from '@deriv/quill-icons'
+import {
+  bulletIconFillVariants,
+  buttonTextVariants,
+  paginationVariants,
+} from './pagination.classnames'
+import { DOTS } from 'hooks/usePaginationRange'
+import { CaptionText } from '..'
+import { PaginationProps } from './pagination.types'
+
+interface PaginationButtonProps<T> extends Pick<PaginationProps<T>, 'variant'> {
+  pageNumber: number | string
+  currentPage: number
+  handleOnClick: (event: string | null) => void
+}
+
+/**
+ * Component to render type of Pagination button based on variant and page number.
+ * @name PaginationButton
+ * @param {number} currentPage - Selected page number
+ * @param {function} handleOnClick - Function to handle click action on the Pagination button
+ * @param {number} pageNumber - Page number
+ * @param {string} variant - Variant of Pagination
+ */
+const PaginationButton = <T,>({
+  currentPage,
+  handleOnClick,
+  pageNumber,
+  variant,
+}: PaginationButtonProps<T>) => {
+  if (variant === 'bullet') {
+    return (
+      <button
+        onClick={() => handleOnClick(String(pageNumber))}
+        className={paginationVariants({ variant })}
+      >
+        <div>
+          <StandaloneCircleFillIcon
+            iconSize="sm"
+            height={12}
+            width={12}
+            className={bulletIconFillVariants({
+              selected: currentPage === pageNumber,
+            })}
```

Comment:
Same for this @likhith-deriv 
```suggestion
            className={bulletIconFillVariants({
              selected: currentPage === pageNumber,
            })}
```